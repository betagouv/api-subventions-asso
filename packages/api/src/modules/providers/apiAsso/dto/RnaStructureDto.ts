export interface RnaStructureDto {
    identite: {
        nom: string;
        sigle: string;
        id_rna: string;
        date_pub_jo: string;
        date_modif_rna: string;
        active: boolean;
        nature: string;
    };
    activites: {
        objet: string;
        lib_objet_social1: string;
        annee_activite_principale: number;
        effectif_salarie_cent: number;
        annee_effectif_salarie_cent: number;
    };
    coordonnees: {
        adresse_siege: {
            cplt_1: string;
            num_voie: number;
            voie: string;
            commune: string;
            cp: number;
            bp?: string;
            type_voie: string;
        };
        adresse_gestion: {
            cplt_1: string;
            voie: string;
            commune: string;
            cp: number;
            pays: string;
        };
    };
}
