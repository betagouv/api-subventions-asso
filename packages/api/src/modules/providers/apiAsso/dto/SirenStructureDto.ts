export interface SirenStructureEtablissementDto {
    actif: boolean;
    adresse: {
        cplt_1?: string;
        num_voie: number;
        voie: string;
        commune: string;
        code_insee: number;
        cp: number;
        type_voie: string;
    };
    id_siret: number;
    est_siege: boolean;
    date_actif: string;
    id_activite_principale: string;
    annee_activite_principale: number;
    effectif_salarie_cent: number;
    id_tranche_effectif?: string;
    annee_effectif_salarie_cent: number;
}

export interface SirenStructureDto {
    identite: {
        nom: string;
        nom_sirene: string;
        id_siren: number;
        id_siret_siege: number;
        id_forme_juridique: number;
        date_creation_sirene: string;
        date_modif_siren: string;
        active: boolean;
        active_sirene: boolean;
        regime: string;
    };
    activites: {
        id_activite_principale: string;
        annee_activite_principale: number;
        id_tranche_effectif: string;
        effectif_salarie_cent: number;
        annee_effectif_salarie_cent: number;
        appartenance_ess: string;
    };
    coordonnees: {
        adresse_siege: {
            cplt_1: string;
            num_voie: number;
            voie: string;
            commune: string;
            code_insee: number;
            cp: number;
            type_voie: string;
        };
        adresse_gestion: {
            cplt_1: string;
            num_voie: number;
            voie: string;
            commune: string;
            code_insee: number;
            cp: number;
            type_voie: string;
        };
    };
    nbEtabsActifs: number;
    etablissement?: SirenStructureEtablissementDto[];
}
