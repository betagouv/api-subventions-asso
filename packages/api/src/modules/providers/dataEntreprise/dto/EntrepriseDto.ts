import EtablisementDto from "./EtablissementDto";

export default interface EntrepriseDto {
    unite_legale: {
        id: number,
        siren:  string,
        statut_diffusion: string,
        unite_purgee?: string,
        date_creation: string,
        sigle?: string,
        sexe?: string,
        prenom_1?: string,
        prenom_2?: string,
        prenom_3?: string,
        prenom_4?: string,
        prenom_usuel?: string,
        pseudonyme?: string,
        identifiant_association: string,
        tranche_effectifs: string,
        annee_effectifs: string,
        date_dernier_traitement: string,
        nombre_periodes: string,
        categorie_entreprise: string,
        annee_categorie_entreprise: string,
        date_fin?: string,
        date_debut: string,
        etat_administratif: string,
        nom?: string,
        nom_usage?: string,
        denomination: string,
        denomination_usuelle_1?: string,
        denomination_usuelle_2?: string,
        denomination_usuelle_3?: string,
        categorie_juridique: string,
        activite_principale: string,
        nomenclature_activite_principale: string,
        nic_siege: string,
        economie_sociale_solidaire: string,
        caractere_employeur: string,
        created_at: string,
        updated_at: string,
        etablissement_siege: {
            id: number,
            siren: string,
            nic: string,
            siret: string,
            statut_diffusion: string,
            date_creation: string,
            tranche_effectifs: string,
            annee_effectifs: string,
            activite_principale_registre_metiers?: string,
            date_dernier_traitement: string,
            etablissement_siege: string,
            nombre_periodes: string,
            complement_adresse: string,
            numero_voie: string,
            indice_repetition?: string,
            type_voie: string,
            libelle_voie: string,
            code_postal: string,
            libelle_commune: string,
            libelle_commune_etranger?: string,
            distribution_speciale?: string,
            code_commune: string,
            code_cedex?: string,
            libelle_cedex?: string,
            code_pays_etranger?: string,
            libelle_pays_etranger?: string,
            complement_adresse_2?: string,
            numero_voie_2?: string,
            indice_repetition_2?: string,
            type_voie_2?: string,
            libelle_voie_2?: string,
            code_postal_2?: string,
            libelle_commune_2?: string,
            libelle_commune_etranger_2?: string,
            distribution_speciale_2?: string,
            code_commune_2?: string,
            code_cedex_2?: string,
            libelle_cedex_2?: string,
            code_pays_etranger_2?: string,
            libelle_pays_etranger_2?: string,
            date_debut: string,
            etat_administratif: string,
            enseigne_1?: string,
            enseigne_2?: string,
            enseigne_3?: string,
            denomination_usuelle?: string,
            activite_principale: string,
            nomenclature_activite_principale: string,
            caractere_employeur: string,
            longitude: string,
            latitude: string,
            geo_score: string,
            geo_type: string,
            geo_adresse: string,
            geo_id: string,
            geo_ligne: string,
            geo_l4: string,
            geo_l5?: string,
            unite_legale_id: number,
            created_at: string,
            updated_at: string
        },
        numero_tva_intra: string,
        etablissements: EtablisementDto[]
    }
}