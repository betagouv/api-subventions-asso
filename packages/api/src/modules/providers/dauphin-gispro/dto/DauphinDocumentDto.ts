/**
 * /!\ This DTO is not complete, because no docs are found, so we complete as we go along.
 */
export default interface DauphinDocumentDto {
    reference: string;
    libelle: { value: string; lang?: string };
    documents: {
        id: string; // this is what we want to download later
        href: string; // this only gets meta-data somehow
        title: string;
        rel: string;
        error?: never;
        expand: {
            properties: {
                "entity:document:date": { value: string }; // using this as "date de dépôt"
                "cmis:creationDate": { value: number };
                "cmis:lastModificationDate": { value: number };
            };
            propertiesExtension: Record<string, unknown>;
        };
    }[];
}

export interface DauphinDocumentFromApplicationDto extends DauphinDocumentDto {
    description: { value: string };
    modeTransmission: string;
    conforme: boolean;
    fonction: string;
    obligatoire: boolean;
    obligatoireSurRecevabilite: boolean;
}
