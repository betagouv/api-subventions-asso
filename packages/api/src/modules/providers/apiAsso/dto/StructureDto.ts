import { Rna } from "dto";

export interface StructureEtablissementDto {
    id_siret: number;
    id_siren: number;
    actif: boolean;
    date_actif: string;
    est_siege: boolean;
    nom?: string;
    telephone?: string;
    courriel?: string;
    note?: string;
    id_activite_principale: string;
    lib_activite_principale: string;
    annee_activite_principale: number;
    id_tranche_effectif: string;
    lib_tranche_effectif: string;
    effectif_salarie_cent: string;
    annee_effectif_salarie_cent: number;
    adresse: {
        cplt_1?: string;
        cplt_2?: string;
        cplt_3?: string;
        num_voie?: number;
        type_voie?: string;
        voie?: string;
        bp?: string;
        cedex?: string;
        cp?: number;
        commune?: string;
        code_insee?: number;
        pays?: string;
    };
}

export interface StructureRibDto {
    titulaire: string;
    banque: string;
    domiciliation: string;
    iban: string;
    bic: string;
    deleted: boolean;
    id_siret: number;
    url?: string;
}

export interface StructureRepresentantLegalDto {
    civilit√©: string;
    nom: string;
    prenom: string;
    fonction: string;
    est_representant_legal: boolean;
    valideur_cec: boolean;
    publication_internet: boolean;
    telephone: string;
    courriel: string;
    deleted: boolean;
    id_siret: number;
}

export interface StructureRnaDocumentDto {
    id: string;
    type: string;
    sous_type: string;
    lib_sous_type: string;
    annee?: number;
    time?: number;
    url: string;
}

export interface StructureDacDocumentDto {
    uuid: string;
    origin: string;
    nom: string;
    commentaire: string;
    annee_validite: number;
    time_depot: string;
    url: string;
    etat: string;
    meta: {
        type: string;
        id_siret: number;
        etat: string;
        iban?: string;
    };
}

export interface StructureDocumentDto {
    asso: {
        documents: DocumentsDto;
    };
}

export interface DocumentsDto {
    nbDocDac: number;
    nbDocRna: number;
    document_dac: StructureDacDocumentDto[];
    document_rna: StructureRnaDocumentDto[];
}

export default interface StructureDto {
    identite: {
        nom: string;
        nom_sirene?: string;
        sigle: string;
        sigle_sirene?: string;
        id_rna: Rna;
        id_ex: string;
        id_siren?: number;
        id_siret_siege: number;
        id_correspondance: string;
        id_forme_juridique: string;
        lib_forme_juridique: string;
        regime: string;
        date_pub_registre?: string;
        volume?: string;
        folio?: string;
        tribunal_instance?: string;
        date_pub_jo?: string;
        date_creat: string;
        date_creation_sirene: string;
        date_dissolution: string;
        date_modif_rna: string;
        date_modif_siren: string;
        nature: string;
        active: boolean;
        util_publique: boolean;
        date_publication_util_publique: string;
        groupement: string;
        domaine: string;
        type_structure_sportive?: string;
        eligibilite_cec: boolean;
        raison_non_eligibilite_cec: string;
        impots_commerciaux: boolean;
    };
    activites: {
        objet: string;
        id_objet_social1: string;
        lib_objet_social1: string;
        id_objet_social2: string;
        lib_objet_social2: string;
        id_activite_principale: string;
        lib_activite_principale: string;
        annee_activite_principale: number;
        champ_action_territorial: string;
        id_tranche_effectif: string;
        lib_tranche_effectif: string;
        effectif_salarie_cent: string;
        annee_effectif_salarie_cent: number;
        appartenance_ess: string;
        date_appartenance_ess: string;
        volontaire_passsport: boolean;
        activites_passsport: string;
        accueil_handicap_moteur: boolean;
        accueil_handicap_mental: boolean;
    };
    coordonnees: {
        adresse_siege: {
            cplt_1?: string;
            cplt_2?: string;
            cplt_3?: string;
            num_voie?: number;
            type_voie?: string;
            voie?: string;
            bp?: string;
            cedex?: string;
            cp?: string;
            commune?: string;
            code_insee?: number;
        };
        adresse_siege_sirene?: {
            cplt_1?: string;
            cplt_2?: string;
            cplt_3?: string;
            num_voie?: number;
            type_voie?: string;
            voie?: string;
            bp?: string;
            cedex?: string;
            cp?: number;
            commune?: string;
            code_insee?: number;
        };
        adresse_gestion?: {
            cplt_1?: string;
            cplt_2?: string;
            cplt_3?: string;
            num_voie?: number;
            type_voie?: string;
            voie?: string;
            bp?: string;
            cedex?: string;
            cp?: number;
            commune?: string;
            code_insee?: number;
            pays?: string;
        };
        telephone?: string;
        courriel?: string;
        site_web?: string;
        publication_internet?: boolean;
    };
    reseau_affiliation: {
        id_rna: string;
        id_siren: number; // not tested since 2023-11-15 change
        nom: string;
        objet: string;
        adresse: string;
        telephone: string;
        courriel: string;
        nb_licencies: number;
        nb_licencies_h: number;
        nb_licencies_f: number;
        numero: string;
        url: string | number;
    }[];
    composition_reseau: {
        nom: string;
        id_rna: string;
        objet: string;
        id_siren: number;
        adresse: string;
    }[];
    adherent_personne_morales: {
        nom: string;
        siret: number;
        id: number;
    }[];
    representant_legal: StructureRepresentantLegalDto[];
    agrement: {
        type: string;
        numero: string;
        niveau: string;
        attributeur: string;
        date_attribution: string;
        url: string | number;
    }[];
    rh: {
        annee: number;
        nb_adherents: number;
        nb_adherents_h: number;
        nb_adherents_f: number;
        nb_benevoles: number;
        nb_volontaires: number;
        nb_salaries: number;
        nb_salaries_etpt: number;
        nb_emplois_aides: number;
        nb_personnels_detaches: number;
        cumul_top_5_salaires: number;
        id_siret: number;
    }[];
    compte: {
        annee: number;
        a_commissaire_aux_comptes: boolean;
        dons: number;
        subv: number;
        subv_cause: string;
        aides_3ans: number;
        charges: number;
        produits: number;
        resultat: number;
        id_siret: number;
    }[];
    etablissement: StructureEtablissementDto[];
    rib: StructureRibDto[];
    commentaire?: {
        note: string;
    };

    document_rna: StructureRnaDocumentDto[];
    document_dac: StructureDacDocumentDto[];
}
