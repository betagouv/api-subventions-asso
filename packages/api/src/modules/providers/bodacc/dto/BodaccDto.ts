import { BodaccRecordDto } from "dto";

export interface BodaccDto {
    total_count: number;
    records: { record: BodaccRecordDto }[];
}
