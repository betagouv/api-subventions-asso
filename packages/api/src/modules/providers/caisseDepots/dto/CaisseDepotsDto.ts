import { Siret } from "dto";

export type OpenDataSoftLinks = { rel: string; href: string };

export interface CaisseDepotsDatasetDto {
    total_count: number;
    links: OpenDataSoftLinks[];
    records: { links: OpenDataSoftLinks[]; record: CaisseDepotsSubventionDto }[];
}

export type CaisseDepotsSubventionDto = {
    id: string;
    timestamp: string;
    size: number;
    fields: {
        montant: number;
        notificationue: "Oui" | "Non";
        objet: string;
        datesversement_debut: string;
        datesversement_fin: string | null;
        dateconvention: string;
        nombeneficiaire: string;
        nature: string;
        nomattribuant: string; // centre financier
        idbeneficiaire: Siret;
        pourcentagesubvention: number;
        conditionsversement: "UNIQUE" | "ECHELONNE";
        idattribuant: Siret;
        referencedecision: string | null; // often null
        idrae: string | null; // often null
    };
};
