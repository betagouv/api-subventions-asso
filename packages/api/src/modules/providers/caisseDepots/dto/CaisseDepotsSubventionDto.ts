import { Siret } from "dto";

export default interface CaisseDepotsSubventionDto {
    id: string;
    timestamp: string;
    fields: {
        montant: number;
        notificationue: "Oui" | "Non";
        objet: string;
        datesversement_debut: string;
        datesversement_fin: string | null;
        dateconvention: string;
        nombeneficiaire: string;
        nature: string;
        nomattribuant: string; // centre financier
        idbeneficiaire: Siret;
        pourcentagesubvention: number;
        conditionsversement: "UNIQUE" | "ECHELONNE";
        idattribuant: Siret;
        referencedecision: string | null; // often null
    };
}
