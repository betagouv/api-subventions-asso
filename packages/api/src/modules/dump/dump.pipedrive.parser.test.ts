import DumpPipedriveParser from "./dump.pipedrive.parser";
import { AgentJobTypeEnum } from "dto";

describe("DumpPipedriveParser", () => {
    describe("adapts", () => {
        it("adapts correctly", () => {
            const headers = [
                "Personne - Nom",
                "Structure",
                "AgentType",
                "DecentralizedLevel",
                "Email",
                "Base - E-mail",
                "Personne - Département",
                "Personne - Région",
                "JobType",
                "Personne - Data.Subvention - Compte actif",
                "Personne - Fonction",
                "Personne - Service",
                "LastName",
                "FirstName",
                "PhoneNumber",
            ];
            const values = [
                [
                    "Prenom NOM",
                    "UNIVERSITE",
                    "Administration déconcentrée",
                    "Départementale",
                    "prenom.nom@base.fr",
                    "prenom.nom",
                    "69 - Rhône",
                    "Auvergne-Rhône-Alpes",
                    "Chef de service,Expert métier",
                    "oui",
                    "",
                    "",
                    "NOM",
                    "Prenom",
                    "06 00 00 00 00",
                ],
                [
                    "Prenom2 NOM2",
                    "UNIVERSITE",
                    "Administration centrale",
                    "Centrale",
                    "prenom2.nom2@base.fr",
                    "prenom2.nom2",
                    "69 - Rhône",
                    "Auvergne-Rhône-Alpes",
                    "",
                    "oui",
                    "",
                    "",
                    "NOM2",
                    "Prenom2",
                    600000000,
                ],
            ];
            // @ts-expect-error test private
            const actual = DumpPipedriveParser.adapts([[headers, ...values]]);
            expect(actual).toMatchSnapshot();
        });
    });

    describe("pathObject adapters", () => {
        describe("jobType adapter", () => {
            it("returns empty array if empty string", () => {
                const toAdapt = undefined;
                const expected = [];
                // @ts-expect-error test private
                const actual = DumpPipedriveParser.pathObject.jobType.adapter(toAdapt);
                expect(actual).toEqual(expected);
            });

            it("returns empty array if undefined", () => {
                const toAdapt = "";
                const expected = [];
                // @ts-expect-error test private
                const actual = DumpPipedriveParser.pathObject.jobType.adapter(toAdapt);
                expect(actual).toEqual(expected);
            });

            it("returns array", () => {
                const toAdapt = "Gestionnaire";
                const expected = [AgentJobTypeEnum.ADMINISTRATOR];
                // @ts-expect-error test private
                const actual = DumpPipedriveParser.pathObject.jobType.adapter(toAdapt);
                expect(actual).toEqual(expected);
            });

            it("returns array with multiple values if coma", () => {
                const toAdapt = "Contrôleur, Expert métier";
                const expected = [AgentJobTypeEnum.CONTROLLER, AgentJobTypeEnum.EXPERT];
                // @ts-expect-error test private
                const actual = DumpPipedriveParser.pathObject.jobType.adapter(toAdapt);
                expect(actual).toEqual(expected);
            });
        });
    });
});
