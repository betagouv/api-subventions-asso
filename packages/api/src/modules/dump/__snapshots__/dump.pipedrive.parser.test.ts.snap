// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DumpPipedriveParser adapts adapts correctly 1`] = `
Array [
  Object {
    "agentType": "DECONCENTRATED",
    "decentralizedLevel": "DEPARTMENTAL",
    "department": "69 - Rh么ne",
    "email": "prenom.nom@base.fr",
    "firstName": "Prenom",
    "jobType": Array [
      "SERVICE_HEAD",
      "EXPERT",
    ],
    "lastName": "NOM",
    "phoneNumber": "06 00 00 00 00",
    "region": "Auvergne-Rh么ne-Alpes",
    "service": "",
    "structure": "UNIVERSITE",
  },
  Object {
    "agentType": "CENTRAL_ADMIN",
    "decentralizedLevel": undefined,
    "department": "69 - Rh么ne",
    "email": "prenom2.nom2@base.fr",
    "firstName": "Prenom2",
    "jobType": Array [],
    "lastName": "NOM2",
    "phoneNumber": "0600000000",
    "region": "Auvergne-Rh么ne-Alpes",
    "service": "",
    "structure": "UNIVERSITE",
  },
]
`;
